examples/src/main/scala/com/cloudera/livy/examples/WordCountApp.scala

```diff
@@ -0,0 +1,243 @@
+  private def getSourcePath(obj: Object): String = {
+    val source = obj.getClass.getProtectionDomain.getCodeSource
+    if (source != null && source.getLocation.getPath != "") {
+      source.getLocation.getPath
+    } else {
+      throw new FileNotFoundException(s"Jar containing ${obj.getClass.getName} not found.")
+    }
+  }
+
+  private def uploadJar(path: String) = {
+    val file = new File(path)
+    val uploadJarFuture = scalaClient.uploadJar(file)
+    Await.result(uploadJarFuture, 40 second) match {
+      case null => println("Successfully uploaded " + file.getName)
+    }
+  }
+
+  /**
+   * Submits a spark streaming job to the livy server.
+   *
+   * The streaming job reads data from the given host and port. The data read
+   * is saved in json format as data frames in the given output path file. If the file is present
+   * it appends to it, else creates a new file. For simplicity, the number of streaming batches
+   * are 2 with each batch for 20 seconds. The Timeout of the streaming job is set to 40 seconds.
+   * @param host Hostname that Spark Streaming context has to connect for receiving data.
+   * @param port Port that Spark Streaming context has to connect for receiving data.
+   * @param outputPath Output path to save the processed data read by the Spark Streaming context.
+   */
+  def processStreamingWordCount(
+      host: String,
+      port: Int,
+      outputPath: String): ScalaJobHandle[Unit] = {
+    scalaClient.submit { context =>
+      context.createStreamingContext(15000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyContent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(12000)
+      ssc.stop(false, true)
+    }
+  }
+
+  /**
+   * Submits a spark sql job to the livy server.
+   *
+   * The sql context job reads data frames from the given json path and executes
+   * a sql query to get the word with max count on the temp table created with data frames.
+   * @param inputPath Input path to the json data containing the words.
+   */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[String] = {
+    scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      try {
+        val rdd = sqlctx.read.json(inputPath)
+        rdd.registerTempTable("words")
+        val result = sqlctx.sql("select word, count(word) as word_count from words " +
+          "group by word order by word_count desc limit 1")
+        result.first().toString()
+      } catch {
+        case exception: IOException => {
+          throw new IOException("The input path does not have any data frame", exception)
```

Comment:
nit: the exception could be for a completely different reason than what the message says. better to just let the exception propagate without trying to guess the reason.